app/models/organization.rb

```diff
@@ -14,6 +14,27 @@ class Organization < ApplicationRecord
 
   auto_strip_attributes :name, squish: true, upcase_first: true
 
+  # ----------------
+  # Region callbacks
+  after_create :create_region
+  before_update :update_region
+
+  def create_region
+    return if region&.persisted?
+    parent = Region.find_by!(region_type: Region.region_types[:root])
+    region = build_region(name: name, description: description, reparent_to: parent)
+    region.region_type = Region.region_types[:organization]
+    region.save!
+  end
+
+  def update_region
+    return unless name_changed? || description_changed?
```

Comment:
Avoid unnecessary updates here if the name or description has not changed.

Reply:
I had this, but removed it because of paranoia. But I think it should be ok to do this...

Reply:
I feel like we can get away with not having to check this, it's one more conditional to keep in mind (there's already too many) while writing all the callbacks.